minimisers/simplex/fMinSearch.m

```diff
@@ -1,66 +1,43 @@
-%   parameters replaced by values in the structure OPTIONS, created
-%   with the OPTIMSET function.  See OPTIMSET for details.  FMINSEARCH uses
-%   these options: Display, TolX, TolFun, MaxFunEvals, MaxIter, FunValCheck,
-%   PlotFcns, and OutputFcn.
-%
-%   X = FMINSEARCH(PROBLEM) finds the minimum for PROBLEM. PROBLEM is a
-%   structure with the function FUN in PROBLEM.objective, the start point
-%   in PROBLEM.x0, the options structure in PROBLEM.options, and solver
-%   name 'fminsearch' in PROBLEM.solver. 
-%
-%   [X,FVAL]= FMINSEARCH(...) returns the value of the objective function,
-%   described in FUN, at X.
-%
-%   [X,FVAL,EXITFLAG] = FMINSEARCH(...) returns an EXITFLAG that describes
-%   the exit condition. Possible values of EXITFLAG and the corresponding
-%   exit conditions are
+%   Reference: Jeffrey C. Lagarias, James A. Reeds, Margaret H. Wright,
+%   Paul E. Wright, "Convergence Properties of the Nelder-Mead Simplex
+%   Method in Low Dimensions", SIAM Journal of Optimization, 9(1):
+%   p.112-147, 1998.
 %
-%    1  Maximum coordinate difference between current best point and other
-%       points in simplex is less than or equal to TolX, and corresponding 
-%       difference in function values is less than or equal to TolFun.
-%    0  Maximum number of function evaluations or iterations reached.
-%   -1  Algorithm terminated by the output function.
+%   Copyright 1984-2018 The MathWorks, Inc.
 %
-%   [X,FVAL,EXITFLAG,OUTPUT] = FMINSEARCH(...) returns a structure
-%   OUTPUT with the number of iterations taken in OUTPUT.iterations, the
-%   number of function evaluations in OUTPUT.funcCount, the algorithm name 
-%   in OUTPUT.algorithm, and the exit message in OUTPUT.message.
+% This is the MATLAB base ``fminsearch`` function adjusted for RAT.
+function [x,fval,exitflag,output] = fMinSearch(funfcn,x,options,dis,varargin)
+% Perform a Nelder-Mead Simplex minimisation of an objective function.
 %
-%   Examples
-%     FUN can be specified using @:
-%        X = fminsearch(@sin,3)
-%     finds a minimum of the SIN function near 3.
-%     In this case, SIN is a function that returns a scalar function value
-%     SIN evaluated at X.
+% Parameters
+% ----------
+% funfcn : function
+%     The function to minimise.
+% x : vector
+%     The starting point to minimise from.
+% options : struct
+%     Controls options for the algorithm.
+% dis : string 
+%     Options for how to display algorithm output.
+% varargin : options
+%     Other options from the Mathworks ``fminsearch``.
 %
-%     FUN can be an anonymous function:
-%        X = fminsearch(@(x) norm(x),[1;2;3])
-%     returns a point near the minimizer [0;0;0].
+% Returns
+% -------
+% x : vector
+%     The minimum value found.
+% fval : vector
+%     The objective function value at output point ``x``.
+% exitflag : integer
+%     Reason ``fimsearch`` stopped: 1 if converged, 0 if max iterations/evals exceeded,
+%     -1 if the output function returned an error.
```

Comment:
Typo "``fMinSearch``". Maybe put these on separate lines as in the removed code, like:

%     Reason ``fMinSearch`` stopped:
%           1 if converged,
%           0 if max iterations/evals exceeded,
%          -1 if the output function returned an error.

Reply:
*Sigh*, so it ignores my tab before each option. Hopefully you get the idea.